presto-native-execution/presto_cpp/main/runtime-metrics/RuntimeMetricsReporter.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/http/HttpServer.h"
+#include "velox/velox/common/base/StatsReporter.h"
+
+namespace facebook::presto {
+
+/// Registers an HTTP GET endpoint to fetch metrics.
+void registerRuntimeMetricsHttpEndpoint(
```

Comment:
After moving it to PrestoServer, this .h file is no longer needed.

Reply:
I agree that we can move this function declaration to PrestoServer.cpp and avoid a header.
It is being invoked only inside PrestoServer.cpp anyway.

Reply:
Done.

Reply:
Sorry about the confusion. What I meant was:

1. Make fetchMetrics() generic in BaseStatsReporter
2. Move both declaration and definition of registerRuntimeMetricsHttpEndpoint() to PrestoServer. Maybe you don't even need a method, just put it in the block where runtime collection is enabled. This way this API becomes a common API for metrics collection.

This way by leveraging polymorphism, you don't need to do the pointer cast.